packages/react-data-grid/src/HeaderGroup.js

```diff
@@ -0,0 +1,136 @@
+const React               = require('react');
+const ReactDOM            = require('react-dom');
+const joinClasses         = require('classnames');
+const ColumnMetrics       = require('./ColumnMetrics');
+const HeaderGroupRow      = require('./HeaderGroupRow');
+const getScrollbarSize    = require('./getScrollbarSize');
+import PropTypes from 'prop-types';
+const createObjectWithProperties = require('./createObjectWithProperties');
+require('../../../themes/react-data-grid-header.css');
+
+type Group = {
+  width: number
+}
+
+// The list of the propTypes that we want to include in the Header div
+const knownDivPropertyKeys = ['height', 'onScroll'];
+
+class HeaderGroup extends React.Component {
```

Comment:
There are parts of `HeaderGroup` that are very similar to `Header` component. Can we extract what is similar to a higher order component to share the functionality?